contracts/schemes/Competition.sol

```diff
@@ -0,0 +1,357 @@
+        proposals[proposalId].rewardSplit = _rewardSplit;
+        proposals[proposalId].startTime = startTime;
+        proposals[proposalId].votingStartTime = _competitionParams[1];
+        proposals[proposalId].endTime = _competitionParams[2];
+        proposals[proposalId].numberOfVotesPerVoters = _competitionParams[3];
+        proposals[proposalId].reputationReward = uint256(_reputationChange);
+        proposals[proposalId].nativeTokenReward = _rewards[0];
+        proposals[proposalId].ethReward = _rewards[1];
+        proposals[proposalId].externalTokenReward = _rewards[2];
+
+        emit NewCompetitionProposal(
+            proposalId,
+            numberOfWinners,
+            proposals[proposalId].rewardSplit,
+            startTime,
+            proposals[proposalId].votingStartTime,
+            proposals[proposalId].endTime,
+            proposals[proposalId].numberOfVotesPerVoters,
+            contributionRewardExt
+        );
+    }
+
+    /**
+    * @dev suggest a competion suggestion
+    * @param _proposalId the proposal id this suggestion is referring to.
+    * @param _descriptionHash a descriptionHash of the suggestion.
+    * @return suggestionId the suggestionId.
+    */
+    function suggest(
+            bytes32 _proposalId,
+            string calldata _descriptionHash
+    )
+    external
+    returns(uint256)
+    {
+      // solhint-disable-next-line not-rely-on-time
+        require(proposals[_proposalId].startTime <= now, "competition not started yet");
+        // solhint-disable-next-line not-rely-on-time
+        require(proposals[_proposalId].endTime > now, "competition ended");
+        suggestionsCounter = suggestionsCounter.add(1);
+        suggestions[suggestionsCounter].proposalId = _proposalId;
+        suggestions[suggestionsCounter].suggester = msg.sender;
+        emit NewSuggestion(_proposalId, suggestionsCounter, _descriptionHash, msg.sender);
+        return suggestionsCounter;
+    }
+
+    /**
+    * @dev vote on a suggestion
+    * @param _suggestionId suggestionId
+    * @return bool
+    */
+    function vote(uint256 _suggestionId)
+    external
+    returns(bool)
+    {
+        bytes32 proposalId = suggestions[_suggestionId].proposalId;
+        require(proposalId != bytes32(0), "suggestion not exist");
+        setSnapshotBlock(proposalId);
+        Proposal storage proposal = proposals[proposalId];
+        // solhint-disable-next-line not-rely-on-time
+        require(proposal.endTime > now, "competition ended");
+        Suggestion storage suggestion = suggestions[_suggestionId];
+        require(suggestion.votes[msg.sender] == 0, "already voted on this suggestion");
+        require(proposal.votesPerVoter[msg.sender] < proposal.numberOfVotesPerVoters,
+        "exceed number of votes allowed");
+        proposal.votesPerVoter[msg.sender] = proposal.votesPerVoter[msg.sender].add(1);
+        suggestion.votes[msg.sender] = suggestion.votes[msg.sender].add(1);
```

Comment:
What is this line for? It is always 1 and always rewritten in the function a few lines below.

Reply:
removed with ced6964